### PR TITLE
Update vic-machine config of external networks

### LINKED ISSUE #3197: vic-machine should not allow configuration of client and management networks when those networks are shared with external
For static IP setups of client and management networks that may share with the external network, the default route on the external network will cover routing for the shared network. 

UPDATE:
Okay, there was a bit of confusion on my part. I thought that we actually bound multiple IP addresses to an interface in the appliance, but we do not. So the planned fix for @stuclem problem is as follows (and I'll change the description to reflect this):

If you have the client and/or management network shared with the external (soon to be public) network then you only configure the external IP and gateway and the configuration of the other "shared" network from vic-machine is not allowed.

So if you have a setup where management, client and external is shared the only networking options available for the VCH are those related to the external network from vic-machine.

If you have management and external shared then the management network options are disabled and you should use the external network configuration.

If you have client and external shared then the external network options should be used and the client network options are disabled.

~~We validate the gateway and ip settings too early in vic-machine and always require both if one is set. This should not be the case.~~

~~external network gateway for external network static ip is required.~~

~~if no management network gateway is set, warn the user only if it is not shared with external.
if no client network gateway is set, warn the user only if it is not shared with external.~~

### PR DESCRIPTION
**Pull request by andrewtchin**
Fixes #3197 
If the user configures a static IP for a network that shares a port group with the public/external network, force the user to configure that IP using the --external-network options to prevent confusion

```
bin/vic-machine-linux create --target root:password@192.168.218.129 --name chin-vic-10 --force --no-tlsverify --client-network-gateway 192.168.218.2/24 --client-network-ip 192.168.218.101/24        
INFO[2016-11-17T12:36:36-08:00] ### Installing VCH ####                      
INFO[2016-11-17T12:36:36-08:00] Using client-network-ip as cname where needed - use --tls-cname to override: 192.168.218.101/24 
INFO[2016-11-17T12:36:36-08:00] Loaded server certificate ./chin-vic-10/server-cert.pem 
WARN[2016-11-17T12:36:36-08:00] Configuring without TLS verify - certificate-based authentication disabled 
INFO[2016-11-17T12:36:36-08:00] Supplied install parameters: --compute-resource { chin-vic-10} --image-store  --volumes map[] --bridge-network chin-vic-10 --client-network { [] {192.168.218.2 ffffff00} {192.168.218.101 ffffff00}} --external-network {VM Network [] {<nil> <nil>} {<nil> <nil>}} --http-proxy <nil> --https-proxy <nil> 
WARN[2016-11-17T12:36:36-08:00] iso file "/home/chin/go/src/github.com/vmware/vic/bin/appliance.iso" version "v0.6.0-0-3aa7acb" inconsistent with installer version "v0.7.0-0-5b61e79" 
WARN[2016-11-17T12:36:36-08:00] iso file "/home/chin/go/src/github.com/vmware/vic/bin/bootstrap.iso" version "v0.6.0-0-3aa7acb" inconsistent with installer version "v0.7.0-0-5b61e79" 
INFO[2016-11-17T12:36:36-08:00] Validating supplied configuration            
INFO[2016-11-17T12:36:36-08:00] Using default datastore: datastore1          
ERRO[2016-11-17T12:36:36-08:00] client network shares port group with public network, but has static IP configuration 
ERRO[2016-11-17T12:36:36-08:00] To resolve this, configure static IP for public network and assign client network to same port group 
ERRO[2016-11-17T12:36:36-08:00] The static IP will be automatically configured for networks sharing the port group 
INFO[2016-11-17T12:36:36-08:00] Configuring static IP for additional networks using port group "VM Network" 
INFO[2016-11-17T12:36:36-08:00] Firewall status: DISABLED on "/ha-datacenter/host/localhost.localdomain/localhost.localdomain" 
INFO[2016-11-17T12:36:36-08:00] Firewall configuration OK on hosts:          
INFO[2016-11-17T12:36:36-08:00] 	"/ha-datacenter/host/localhost.localdomain/localhost.localdomain" 
WARN[2016-11-17T12:36:36-08:00] Evaluation license detected. VIC may not function if evaluation expires or insufficient license is later assigned. 
INFO[2016-11-17T12:36:36-08:00] License check OK                             
INFO[2016-11-17T12:36:36-08:00] DRS check SKIPPED - target is standalone host 
ERRO[2016-11-17T12:36:36-08:00] --------------------                         
ERRO[2016-11-17T12:36:36-08:00] Static IP on network sharing port group with public network - Configuration ONLY allowed through public network options 
ERRO[2016-11-17T12:36:36-08:00] Create cannot continue: configuration validation failed 
ERRO[2016-11-17T12:36:36-08:00] --------------------                         

```

### CODE REVIEW

File: lib/install/validate/network.go [hunk 1]
```diff
@@ -201,12 +203,33 @@ func (v *Validator) network(ctx context.Context, input *data.Data, conf *config.
...
 	v.NoteIssue(err)
 
+	// client and management networks need to have at least one
+	// routing destination if gateway was specified
+	for nn, n := range map[string]*data.NetworkConfig{
+		"client":     &input.ClientNetwork,
+		"management": &input.ManagementNetwork,
+	} {
+		if n.Name == input.ExternalNetwork.Name {
+			// no Destinations requried if sharing with PublicNetwork
```
Comment (andrewtchin) [inline lib/install/validate/network.go]: lol

**Comment by hmahmood** [comment]
lgtm

**Comment by matthewavery** [comment]
lgtm

**Comment by andrewtchin** [comment]
♻️

**Pull request merged**